nilearn/glm/contrasts.py

```diff
@@ -106,6 +107,9 @@ def compute_contrast(labels, regression_result, con_val, contrast_type=None):
         for label_ in regression_result:
             label_mask = labels == label_
             reg = regression_result[label_]
+            con_val = pad_contrast(
+                con_val=con_val, theta=reg.theta, contrast_type=contrast_type
+            )
```

Comment:
@bthirion 
any reason why for F contrasts we do not rely on the FContrast methods and do something like 

```python
reg = regression_result[label_].Fcontrast(con_val)
```

the way we do it for T contrasts above?

Reply:
IIRC, this is because we had to trick it somehow to support fixed effects on F contrasts.

Reply:
ok this may require a comment if we can remember the why otherwise I could add a TODO to try to refactor this

Reply:
I can add this to your PR.

Reply:
Added a TODO for now in case we want to merge and tackle this in a separate PR.

Reply:
Maybe rather in another PR.